### PR TITLE
INT-2416: Clarify copyHeaders in s-a.adoc

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-2416

### CODE REVIEW

**Comment by garyrussell** [comment]
Rebased, + polishing.

File: src/reference/asciidoc/service-activator.adoc [hunk 1]
```diff
@@ -166,12 +166,13 @@ An `ErrorMessage` is sent to the `errorChannel` message header, if present. Othe
...
-So, if you try to perform some existing headers removal in your service method, they are going to appear in the reply message back.
-This, essentially, not a Service Activator responsibility to remove headers for reply message and pursuing the loosely-coupling principle it is a proper way to pair with a `HeaderFilter` in the integration flow.
-The Transformer can be used in a Service Activator role, but returning a full `Message<?>` from method call is already an end-user responsibility and you need to ensure the proper headers population for a reply message, including copying request message headers.
+This works the same way for most Spring Integration `MessageHandler` implementations, when interaction is based on a POJO method invocation.
+
+A complete `Message<?>` object can also be returned from the method.
+However keep in mind that, unlike <<transformer transformers>>, for a Service Activator this message will be modified by copying the headers from the request message if they are not already present in the returned message.
```
Comment (artembilan) [inline src/reference/asciidoc/service-activator.adoc]: I think we need a comma after `transformer`

File: src/reference/asciidoc/service-activator.adoc [hunk 2]
```diff
@@ -166,12 +166,13 @@ An `ErrorMessage` is sent to the `errorChannel` message header, if present. Othe
...
+So, if your method parameter is a `Message<?>` and you copy some, but not all, existing headers in your service method, they will reappear in the reply message.
+It is not a Service Activator responsibility to remove headers from a reply message and, pursuing the loosely-coupled principle, it is better to add a `HeaderFilter` in the integration flow.
+Alternatively, a Transformer can be used instead of a Service Activator but, in that case, when returning a full `Message<?>` the method is completely responsible for the message, including copying request message headers (if needed).
+It is important to ensure that important framework headers (e.g. `replyChannel`, `errorChannel`), if present, must be preserved.
```
Comment (artembilan) [inline src/reference/asciidoc/service-activator.adoc]: Some kind of tautology with the `important` word...

**Review by artembilan — COMMENTED** [review]
Cool! Thank you for polishing!

I left a couple comments on your commit

File: src/reference/asciidoc/service-activator.adoc (hunk 2 above)
Comment (garyrussell) [reply src/reference/asciidoc/service-activator.adoc]: `You must ensure that important framework headers (e.g. `replyChannel`, `errorChannel`), if present, are preserved in the reply message.`.

**Comment by garyrussell** [comment]
Merged as 0c14aa052ba44aa8a86298b563c58c8b334dbbf0

**Pull request closed without merging**